python/pyiceberg/io/pyarrow.py

```diff
@@ -492,11 +710,7 @@ def _file_to_table(
         schema_raw = None
         if metadata := parquet_schema.metadata:
             schema_raw = metadata.get(ICEBERG_SCHEMA)
-        if schema_raw is None:
-            raise ValueError(
-                "Iceberg schema is not embedded into the Parquet file, see https://github.com/apache/iceberg/issues/6505"
-            )
-        file_schema = Schema.parse_raw(schema_raw)
+        file_schema = Schema.parse_raw(schema_raw) if schema_raw is not None else pyarrow_to_schema(parquet_schema)
```

Comment:
If the `field-id`'s are not set, I believe there is another fallback level where we look at the names and try to look them up in the table schema. That should be done in a separate PR, but maybe good to add a comment here.

Reply:
Added a TODO comment here. Just curious, will the fallback level be something like the NameMapping feature?